pkg/command/buildbase.go

```diff
@@ -93,6 +109,14 @@ func (opts *BaseBuildOptions) Validate(cmd *cobra.Command, args []string) error
 	return nil
 }
 
+// NamespaceOrDefault returns the configured namespace or the default kubernetes namespace
+func (opts *BaseBuildOptions) NamespaceOrDefault() string {
+	if opts.Namespace != "" {
+		return opts.Namespace
+	}
+	return Namespace()
+}
```

Comment:
I think that in `Validate()` we should be able to `opts.Namespace = viper.GetString("namespace")`, and if that's empty fallback on `opts.Namespace = Namespace()`.  Then we can just rely on this being set downstream.

Reply:
though I've always found viper a bit tricky to use in tests given its global/static/env-var dependency; would we only invoke viper if it wasn't already defined right?